run_desktop_tests.sh

```diff
@@ -15,124 +15,104 @@
-cd ../..
-cd tools/heapviz
-cargo fmt -- --check
-cd ../..
-cd bootloader
-cargo fmt -- --check
-cd ..
-
-echo "Running Clippy lints..."
-cargo clippy --lib --tests --bins --benches --features std -- -D warnings
-cargo clippy --lib --tests --bins --benches --features std,with_ctap1,ed25519,vendor_hid -- -D warnings
-cargo clippy --lib --tests --bins --benches --features std,with_ctap1,with_nfc,ed25519,vendor_hid -- -D warnings
+./fuzzing_setup.sh
+# Excludes std
+MOST_FEATURES=config_command,debug_allocations,debug_ctap,panic_console,verbose,with_ctap1,vendor_hid,ed25519
 
-echo "Building sha256sum tool..."
-cargo build --manifest-path third_party/tock/tools/sha256sum/Cargo.toml
-echo "Checking that heapviz tool builds properly..."
-cargo build --manifest-path tools/heapviz/Cargo.toml
-echo "Testing heapviz tool..."
-cargo test --manifest-path tools/heapviz/Cargo.toml
-
-echo "Checking that CTAP2 builds properly..."
+echo "Checking that OpenSK builds properly..."
 cargo check --release --target=thumbv7em-none-eabi
-cargo check --release --target=thumbv7em-none-eabi --features with_ctap1
-cargo check --release --target=thumbv7em-none-eabi --features vendor_hid
-cargo check --release --target=thumbv7em-none-eabi --features ed25519
+cargo check --release --target=thumbv7em-none-eabi --features config_command
+cargo check --release --target=thumbv7em-none-eabi --features debug_allocations
 cargo check --release --target=thumbv7em-none-eabi --features debug_ctap
 cargo check --release --target=thumbv7em-none-eabi --features panic_console
-cargo check --release --target=thumbv7em-none-eabi --features debug_allocations
 cargo check --release --target=thumbv7em-none-eabi --features verbose
-cargo check --release --target=thumbv7em-none-eabi --features debug_ctap,with_ctap1
-cargo check --release --target=thumbv7em-none-eabi --features debug_ctap,with_ctap1,vendor_hid,ed25519,panic_console,debug_allocations,verbose
-
-echo "Checking that examples build properly..."
+cargo check --release --target=thumbv7em-none-eabi --features with_ctap1
+cargo check --release --target=thumbv7em-none-eabi --features with_nfc
+cargo check --release --target=thumbv7em-none-eabi --features vendor_hid
+cargo check --release --target=thumbv7em-none-eabi --features ed25519
+cargo check --release --target=thumbv7em-none-eabi --features rust_crypto
+cargo check --release --target=thumbv7em-none-eabi --features "$MOST_FEATURES"
 cargo check --release --target=thumbv7em-none-eabi --examples
 cargo check --release --target=thumbv7em-none-eabi --examples --features with_nfc
+cargo check --release --target=thumbv7em-none-eabi --manifest-path bootloader/Cargo.toml
+cargo check --release --manifest-path tools/heapviz/Cargo.toml
 
-echo "Checking that bootloader builds properly..."
-cd bootloader
-cargo check --release --target=thumbv7em-none-eabi
-cd ..
+echo "Checking Rust formatting..."
+cargo fmt -- --check
+cargo +nightly fmt --manifest-path libraries/opensk/Cargo.toml -- --check
```

Comment:
Here we rely on the nightly toolchain to be installed. This happens to be the case on Github and locally for most, but it's not necessarily true. We would need some `rustup install nightly` in `setup.sh`.

And I guess we need nightly because the old nightly has problems.

Reply:
`rustup install nightly` is in `setup.sh`. Do we also need some components, or just an oversight?

Reply:
And we don't need nightly anymore outside of testing, I think. It's just that I want to test all libraries with the latest clippy etc, to get early warnings when anything breaks.

Reply:
My bad, I just checked for `rustup show` and assumed there wouldn't be more. All good.